price-feeder/oracle/client/chain_height.go

```diff
@@ -102,5 +103,10 @@ func (chainHeight *ChainHeight) GetChainHeight() (int64, error) {
 	chainHeight.mtx.RLock()
 	defer chainHeight.mtx.RUnlock()
 
+	if chainHeight.lastChainHeight == 0 &&
+		chainHeight.errGetChainHeight == nil {
```

Comment:
I think the condition is wrong, seams that once the err is set it will never be unset. Also why checking `lastChainHeight == 0`? since the chain is running and already above 0, then we will never set `chainHeight.errGetChainHeight`

Reply:
This condition occurs when the tick runs before a new height is received from the subscription. Alternatively we could just fill the height with a query when we initialize the object so that this does not happen.

Reply:
we should initialize with the right value , rather than zero